### PR TITLE
Setup aws s3

### PR DESCRIPTION
**Pull request by GurkirtSingh**
### What this PR does (required):
- setup Amazon S3
- creates route to upload profile picture `/upload/profile-pic`

### Any information needed to test this feature (required):
- This feature required new **_environmental variables_** which will be be shared on slack

### Any issues with the current functionality (optional):
- Currently, there is file filter applied to multer object to detected if uploaded file is an image, but there is no feedback to user about wrong file.
- file extension of  uploaded file is hard coded to `.jpg`

### CODE REVIEW

File: server/cloud storage/amazonS3.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const secretAccessKey = process.env.AWS_SECRET_KEY;
+
+const creds = new AWS.Credentials(accessKeyId, secretAccessKey);
+
+exports.s3 = new AWS.S3({
+  region,
+  credentials: creds,
+});
+
+// // upload file to bucket
```
Comment (rajivtitus) [inline server/cloud storage/amazonS3.js]: Please remove commented code or uncomment

File: server/controllers/upload.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        cb(null, filename);
+      },
+    }),
+    fileFilter: imageFilter,
+  });
+
+exports.uploadProfilePic = asyncHandler(async (req, res, next) => {
+  // get the profile
+  const profile = await Profile.findOne({ userId: req.user.id });
+  if (!profile) {
```
Comment (rajivtitus) [inline server/controllers/upload.js:33]: 👍🏽

File: server/controllers/upload.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    res.status(404);
+    throw new Error("Profile doesn't exist");
+  }
+
+  //get the image from request
```
Comment (rajivtitus) [inline server/controllers/upload.js]: We can remove this comment

File: server/controllers/upload.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const filename = `${req.user.id}-profile-pic.jpg`;
+  const bucket = process.env.AWS_BUCKET_NAME;
+
+  try {
+    // upload image
```
Comment (rajivtitus) [inline server/controllers/upload.js]:
```suggestion
```

Remove this comment too

File: server/controllers/upload.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    const uploadSingle = upload(bucket, filename).single("image");
+
+    uploadSingle(req, res, (err) => {
+      if (err) return res.status(400).json(err);
+
+      profile.set({ photo: req.file.location });
+      profile.save().then((profile) => {
+        res.status(201);
```
Comment (rajivtitus) [inline server/controllers/upload.js]:
```suggestion
        res.status(200);
```
A `200` status code seems more appropriate here as we're only updating the Profile model (not creating a new resource)

File: server/controllers/upload.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      },
+      key: function (req, file, cb) {
+        cb(null, filename);
+      },
+    }),
+    fileFilter: imageFilter,
+  });
+
+exports.uploadProfilePic = asyncHandler(async (req, res, next) => {
+  // get the profile
```
Comment (rajivtitus) [inline server/controllers/upload.js]:
```suggestion
```

File: server/controllers/upload.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      if (err) return res.status(400).json(err);
+
+      profile.set({ photo: req.file.location });
+      profile.save().then((profile) => {
+        res.status(201);
+        res.json({ profile });
+      });
+    });
+  } catch (err) {
+    res.status(404);
```
Comment (rajivtitus) [inline server/controllers/upload.js]:
```suggestion
    res.status(500);
```

**Review by rajivtitus — COMMENTED** [review]
Very good work with this! 💯 
Just some minor cleanup comments and this should be good to go 👍🏽

File: server/cloud storage/amazonS3.js (hunk 1 above)
Comment (GurkirtSingh) [reply server/cloud storage/amazonS3.js]: I have already removed this in integration PR.

**Pull request merged**